opencti-platform/opencti-front/src/components/ProgressDialog.tsx

```diff
@@ -0,0 +1,165 @@
+    private batchingCompleted: boolean = false,
+    private batchingCancelled: boolean = false,
+    private errorCount: number = 0,
+    private successCount: number = 0,
+  ) {}
+
+  public getCurrentIncrement(): number {
+    return this.currentIncrement;
+  }
+
+  public getCurrentMaxIncrement(): number {
+    return this.currentMaxIncrement;
+  }
+
+  public getBatchingCompleted(): boolean {
+    return this.batchingCompleted;
+  }
+
+  public getBatchingCancelled(): boolean {
+    return this.batchingCancelled;
+  }
+
+  public getSuccessCount(): number {
+    return this.successCount;
+  }
+
+  public getErrorCount(): number {
+    return this.errorCount;
+  }
+
+  public resetCurrentIncrement(): number {
+    this.currentIncrement = 0;
+    return this.currentIncrement;
+  }
+
+  public resetSuccessCount(): number {
+    this.successCount = 0;
+    return this.successCount;
+  }
+
+  public resetErrorCount(): number {
+    this.errorCount = 0;
+    return this.errorCount;
+  }
+
+  public resetCurrentMaxIncrement(incrementValue: number): number {
+    this.currentMaxIncrement = incrementValue;
+    return this.currentMaxIncrement;
+  }
+
+  public setCurrentIncrement(incrementValue: number): number {
+    this.currentIncrement += incrementValue;
+    return this.currentIncrement;
+  }
+
+  public setCurrentMaxIncrement(incrementMaxValue: number): number {
+    this.currentMaxIncrement = incrementMaxValue;
+    return this.currentMaxIncrement;
+  }
+
+  public setBatchingCompleted(batchingFlag = false): boolean {
+    this.batchingCompleted = batchingFlag;
+    return this.batchingCompleted;
+  }
+
+  public setBatchingCancelled(batchingFlag = false): boolean {
+    this.batchingCancelled = batchingFlag;
+    return this.batchingCancelled;
+  }
+
+  public updateSuccessCount(incrementValue: number): number {
+    this.successCount += incrementValue;
+    return this.successCount;
+  }
+
+  public updateErrorCount(incrementValue: number): number {
+    this.errorCount += incrementValue;
+    return this.errorCount;
+  }
+}
+
+type ProgressDialogProps = {
+  openProgressDialog: boolean
+  handleClickCloseProgress: () => void
+};
+
+type LinearProgressClasses = {
+  root: string
+};
+
+type LinearProgressProps = {
+  classes: LinearProgressClasses
+  variant?: 'determinate' | 'indeterminate' | 'buffer' | 'query';
+  value: number
+};
+
+const LinearProgressWithLabel = (props: LinearProgressProps) => {
+  return (
+    <div style={{ display: 'flex', alignItems: 'center' }}>
+      <div style={{ width: '100%', marginRight: 1 }}>
+        <LinearProgress {...props} />
+      </div>
+      <div style={{ minWidth: 35 }}>
+        <Typography variant="body2" color="text.secondary">{`${Math.round(
+          props.value,
+        )}%`}</Typography>
+      </div>
+    </div>
+  );
+};
+
+const useStyles = makeStyles(() => ({
+  progress: {},
+}));
+
+export const progressDialogStats = new ProgressDialog(0);
```

Comment:
Using a global variable like this cannot work. If you instantiate multiple progress bars in the application, they will share the same instance of this class.

As explain above (comment on the class). The state of the progress bar should be kept locally inside the parent component and pass through the progress bar component using its props

Reply:
Out of scope. Will be address later.